test/e2e/app-dir/rewrites-redirects/rewrites-redirects.test.ts

```diff
@@ -0,0 +1,82 @@
+import { createNextDescribe } from 'e2e-utils'
+import { waitFor } from 'next-test-utils'
+
+createNextDescribe(
+  'redirects and rewrites',
+  {
+    files: __dirname,
+    dependencies: {
+      react: 'latest',
+      'react-dom': 'latest',
+      typescript: 'latest',
+      '@types/react': 'latest',
+      '@types/node': 'latest',
+    },
+  },
+  ({ next }) => {
+    /**
+     * All test will use a link/button to navigate to '/*-before' which should be redirected by correct redirect/rewrite to '/*-after'
+     */
+    describe.each(['link', 'button'])('navigation using %s', (testType) => {
+      it('should rewrite from middleware correctly', async () => {
+        const browser = await next.browser('/')
+        browser.elementById(`${testType}-middleware-rewrite`).click()
+        await waitFor(200)
```

Comment:
These need to waitForElementByCss instead

Reply:
Migrated